incubator/virtualcluster/pkg/syncer/controllers/configmap/controller.go

```diff
@@ -0,0 +1,130 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package configmap
+
+import (
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/api/core/v1"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+
+	"k8s.io/klog"
+
+	"github.com/multi-tenancy/incubator/virtualcluster/pkg/syncer/reconciler"
+	sc "github.com/multi-tenancy/incubator/virtualcluster/pkg/syncer/controller"
+	"github.com/multi-tenancy/incubator/virtualcluster/pkg/syncer/listener"
+	"github.com/multi-tenancy/incubator/virtualcluster/pkg/syncer/cluster"
+	"github.com/multi-tenancy/incubator/virtualcluster/pkg/syncer/conversion"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+type controller struct {
+	configMapClient                 v1core.ConfigMapsGetter
+	multiClusterConfigMapController *sc.MultiClusterController
+}
+
+func Register(
+	configMapClient v1core.ConfigMapsGetter,
+	configMapInformer coreinformers.ConfigMapInformer,
+) {
+	c := &controller{
+		configMapClient: configMapClient,
+	}
+
+	options := sc.Options{Reconciler: c}
+	multiClusterConfigMapController, err := sc.NewController("tenant-masters-configmap-controller", options)
+	if err != nil {
+		klog.Errorf("failed to create multi cluster configmap controller %v", err)
+	}
+	c.multiClusterConfigMapController = multiClusterConfigMapController
+
+	configMapInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: c.backPopulate,
+		},
+	)
+
+	// add controller as listener
+	listener.AddListener(c)
+
+	go multiClusterConfigMapController.Start(wait.NeverStop)
+}
+
+func (c *controller) backPopulate(obj interface{}) {
+	configMap := obj.(*v1.ConfigMap)
+	clusterName, namespace := conversion.GetOwner(configMap)
+	if len(clusterName) == 0 {
+		return
+	}
+	_, err := c.multiClusterConfigMapController.Get(clusterName, namespace, configMap.Name)
+	if errors.IsNotFound(err) {
+		return
+	}
+}
+
+func (c *controller) Reconcile(request reconciler.Request) (reconciler.Result, error) {
+	klog.Infof("reconcile configmap %s/%s %s event for cluster %s", request.Namespace, request.Name, request.Event, request.Cluster.Name)
+
+	switch request.Event {
+	case reconciler.AddEvent:
+	case reconciler.UpdateEvent:
+		err := c.reconcileConfigMapUpdate(request.Cluster.Name, request.Namespace, request.Name, request.Obj.(*v1.ConfigMap))
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+	case reconciler.DeleteEvent:
+		err := c.reconcileConfigMapRemove(request.Cluster.Name, request.Namespace, request.Name, request.Obj.(*v1.ConfigMap))
+		if err != nil {
+			return reconciler.Result{}, err
+		}
+	}
+	return reconciler.Result{}, nil
+}
+
+func (c *controller) reconcileConfigMapUpdate(cluster, namespace, name string, configMap *v1.ConfigMap) error {
+	targetNamespace := cluster + "-" + namespace;
```

Comment:
take this conversion into the `conversion` package? this function would be referred around the vc project.
redundant `;`. please do `go fmt` before submit PR.

Reply:
done.